java/kotlin-extractor/src/main/kotlin/KotlinFileExtractor.kt

```diff
@@ -4814,12 +4921,20 @@ open class KotlinFileExtractor(
         }
     }
 
-    private inner class DeclarationStackAdjuster(declaration: IrDeclaration): Closeable {
+    private inner class DeclarationStackAdjuster(val declaration: IrDeclaration, val overriddenAttributes: OverriddenFunctionAttributes? = null): Closeable {
```

Comment:
Wouldn't it make more sense to push `(declaration, overriddenAttributes)` onto the stack? That way we wouldn't have to maintain two sets of states (`overriddenFunctionAttributes` and `declarationStack`)?

Reply:
I separated them because I needed the function override state to be propagated when extracting generic type instantiations as mentioned above and didn't want to get into the decl stack being non-empty on completing extraction of that type, but I can keep them together if that's preferred at the slight cost of searching the stack.